wakatime/riscv64.patch

```diff
@@ -0,0 +1,34 @@
+--- PKGBUILD
++++ PKGBUILD
+@@ -14,10 +14,19 @@ license=('BSD')
+ depends=('glibc')
+ makedepends=('git' 'go')
+ checkdepends=('svn')
+-source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}")
+-sha256sums=('SKIP')
++source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}"
++        "main_test-arch.patch::https://github.com/wakatime/wakatime-cli/pull/645.patch")
++sha256sums=('SKIP'
++            'a53e0d001d2a8e0bcf1e7c0ceb82a82d0ece4b65663af56387d898fb5b2e1e83')
+ 
+-_binname="${pkgname}-cli-linux-amd64"
++_GOARCH=$(go env GOARCH)
++_binname="${pkgname}-cli-linux-${_GOARCH}"
```

Comment:
Hi, it appears that `go` has not been installed when this line is executed, which results in `_GOARCH` being an empty string. Do you have any suggestions for a solution?

Reply:
Can we place these lines in `build()`?

Reply:
Thanks!

Reply:
Actually I think it might be better to initialize the enviroment variable in `prepare()`?

Reply:
Actually I think it might be better to initialize the enviroment variable in `prepare()`?

Reply:
Hi @r-value, by Arch Linux wiki page [Creating packages](https://wiki.archlinux.org/title/Creating_packages#prepare()), `prepare()` is used to prepare sources and will not be run if `--noextract` is specified for `makepkg`. However, in this case, the variable `_binname` should always be set regardless of whether the source code is extracted.

> **prepare()**
>
> With this function, commands that are used to prepare sources for building are run, such as [patching](https://wiki.archlinux.org/title/Patching_in_ABS). This function runs right after package extraction, before [pkgver()](https://wiki.archlinux.org/title/Creating_packages#pkgver()) and the build function. If extraction is skipped (makepkg --noextract), then prepare() is not run.

Reply:
> Hi @r-value, by Arch Linux wiki page [Creating packages](https://wiki.archlinux.org/title/Creating_packages#prepare()), `prepare()` is used to prepare sources and will not be run if `--noextract` is specified for `makepkg`. However, in this case, the variable `_binname` should always be set regardless of whether the source code is extracted.
> 
> > **prepare()**
> > With this function, commands that are used to prepare sources for building are run, such as [patching](https://wiki.archlinux.org/title/Patching_in_ABS). This function runs right after package extraction, before [pkgver()](https://wiki.archlinux.org/title/Creating_packages#pkgver()) and the build function. If extraction is skipped (makepkg --noextract), then prepare() is not run.

True enough.